core/templates/components/rubrics-editor/rubrics-editor.component.ts

```diff
@@ -91,8 +91,30 @@ export class RubricsEditorComponent {
     this.explanationEditorIsOpen[difficulty][index] = true;
   }
 
-  isExplanationValid(difficulty: string, index: number): boolean {
-    return Boolean(this.editableExplanations[difficulty][index]);
+  isExplanationLengthValid(difficulty: string, idx: number): boolean {
+    return this.editableExplanations[difficulty][idx].length <= 300;
+  }
+
+  isMediumLevelExplanationValid(): boolean {
+    if (this.rubrics[1] === undefined) {
+      return false;
+    }
+    return (this.rubrics[1]._explanations.length >= 1);
```

Comment:
1. What exactly is "medium level" explanation? Have we used this terminology anywhere else?
2. Instead of relying on the index, is there a way to determine if a rubric is "medium level" or not? Relying on indices can lead to confusion for other developers reading the code and are usually error prone.
3. Can we use getters in the Rubric model instead of directly accessin private variables (i.e. starting with `_`)?

Reply:
1. So in rubrics, we have 3 levels right? Easy, Medium, and Hard. Each level may have explanations and it's mandatory to have an explanation for the medium level(after this PR). I'm not sure if we have used this terminology anywhere else, if we need to change this then please do let me know.
2. I'm not sure if we'll be able to do that because we are getting rubrics from the parent component as an input and it's an array of all the information related to all the 3 levels of rubrics and on the basis of indexes I'm accessing them, please do let me know if we can do this other way around.
3. By this do you mean that I should create a function in which I should get the values like I'm getting here that way I'll not be accessing the private functionality directly?

Please do let me know about these changes and I'll perform them ASAP.
Thanks @kevintab95

Reply:
Done, Please do let me know if this approach is okay.
Thanks